frontend/src/common/utils/checkForRevisionChange.ts

```diff
@@ -78,6 +110,7 @@ function checkDatasetsForChanges(
   // Check dataset fields for differences
   return Array.from(publishedDatasets.values()).some((publishedDataset) => {
     const revisedDataset = revisedDatasets.get(publishedDataset.id) as Dataset;
+    if (revisedDataset == undefined) return true;
```

Comment:
I am not an expert but I recall that using the triple equal should be the correct behavior here